configure.ac

```diff
@@ -558,6 +558,7 @@ src/Makefile
 src/cli/Makefile
 src/ncp/Makefile
 src/core/Makefile
+src/diag/Makefile
```

Comment:
Should we introduce a build flag here? What if someone does not want to include diag but only wants a bare CLI or bare NCP?